src/main/java/org/truffleruby/core/adapters/OutputStreamAdapter.java

```diff
@@ -34,7 +34,7 @@ public OutputStreamAdapter(RubyContext context, DynamicObject object, Encoding e
     @Override
     public void write(int bite) throws IOException {
         context.send(object, "write", null,
-                StringOperations.createString(context, RopeBuilder.createRopeBuilder(new byte[]{(byte) bite}, encoding).toRope(CodeRange.CR_VALID)));
+                StringOperations.createString(context, RopeOperations.create((byte) bite, encoding, CodeRange.CR_UNKNOWN)));
```

Comment:
I wonder if the value of `bite` can be outside the `byte` range.

Reply:
That's in the contract of OutputStream so we should be fine.

Reply:
We can add an assertion if you'd like.

Reply:
I think it's fine since this is only called by the OutputStreamWriter which explicitly converts characters to bytes.